### ISSUE TITLE
Guide placement is incorrect when using Source Code Pro font 

### ISSUE THREAD

**Issue by dwittenb** [grosenberg/IndentGuide#20]
I am using Eclipse for C/C++ Version: 2022-12 (4.26.0). 
In  "Window - Preferences - General - Editors - Text Editors" the "Displayed tab width" is set to 4. 
In  "Window - Preferences - C/C++ - Code Style - Formatter - Edit - Indentation" the Tab size is set to 4 too. 

In the Editor window the "Indent guide" shows the Indentation at character position 8, 16, 24 and so on. I expect that Indentation guide shows the Indentation on position 4, 8, 12, 16 and so on.

I reinstalled the plugin an the eclipse IDE, no change. I tryed another font and different size. I tried to change settings of Width and Shift and Drawing options in "Window - Preferences - Editors - Text Editors - Indent Guide".
Nothing worked for me.
What is wrong with my settings.

**Comment by grosenberg** [comment]
Are you using display scaling? If so, what percent?

What OS?

**Comment by dwittenb** [comment]
I am using display scaling with 150%. 
I am running WIN11 on a UHD Monitor with 3840 x 2160.
In the eclipse.ini I have added these three lines, to get the icons in a readable size:

```
-Dswt.enable.autoScale=true
-Dswt.autoScale=200
-Dswt.autoScale.method=nearest
```

I am using the eclipse named sloeber with the same ini-settings under WIN11 and on the UHD-monitor with scaling 150%.  I have installed the "Indent Guide" too and everything works fine.

Next I will try the pre-releease v1.8.0 with Eclipse for C/C++.

**Comment by dwittenb** [comment]
pre-releease v1.8.0 installed.
The indentation is better but still not good. 
It looks that indentation is placed a little bit before the tab-size.
In the Indent Guide settings shift is set to 8 but it still seems that on each tab the distance before the tab increases, IIt looks like the shift is not enough, but no value greater than 8 is possible.
![Screenshot 2023-02-09 232104](https://user-images.githubusercontent.com/2135322/217952766-12cd701f-89b1-430a-83fe-1a88760400ba.jpg)

**Comment by dwittenb** [comment]
Perhaps the problem is that indent Guide does not take the font metrics into account.
Until now I was using a monospaced font "source code pro" with size 10.
When changing to "Courier New" with size 10 and setting shift to 2 everything seems to work, have a look at the screenshot !!!
![Screenshot 2023-02-09 233410](https://user-images.githubusercontent.com/2135322/217954533-50a037e6-f682-4087-ab6e-7744e482bd66.jpg)

**Comment by grosenberg** [comment]
Why Source Code Pro fails is unknown. All other Mono (& Code) fonts tried seem to work correctly. Also, appears to work correctly for most proportional fonts, so the plugin is respecting 'correct' font metrics.

Fira Code works correctly, is visually similar to Source Code Pro, and includes ligatures.

**Comment by grosenberg** [comment]
Problem lies with the font reporting the wrong metrics.

**Comment by grosenberg** [comment]
Fixed in release v 2.2.2
